lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -74,9 +74,9 @@ def post_notes(app_id, release_id, release_notes)
       end
 
       def get_upload_token(app_id, binary_path)
-        begin
+        if !binary_path.nil? && File.exist?(binary_path)
           binary_hash = Digest::SHA256.hexdigest(File.open(binary_path).read)
-        rescue Errno::ENOENT
+        else
```

Comment:
In this case, where there is a specific error case to check before moving on, I would avoid the `else` altogether:

```
if (error case)
  crash!
end

binary_hash = thing
```

In this case it also takes the assignment out of the conditional, which is nice.